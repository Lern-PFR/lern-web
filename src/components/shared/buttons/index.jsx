import PropTypes from 'prop-types';
import {
	primary,
	standard,
	subtle,
	dropdown,
	danger,
	iconButton,
	floatingActionButton,
	primaryLink,
	outlined,
	subtleLink,
} from 'theme/buttonStyles';
import DynamicButtonComponent from './DynamicButtonComponent';
import DynamicLinkButtonComponent from './DynamicLinkButtonComponent';

// @TODO: write link and subtleLink buttons once navigation has been set up.

/**
 * @name PrimaryButton
 * @description Button component to use as Call to action. Should only be used once per page (header not included).
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const PrimaryButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent {...primary} {...otherProps}>
		{children}
	</DynamicButtonComponent>
);

/**
 * @name PrimaryLinkButton
 * @description Link button component to use as Call to action. Should only be used once per page (header not included).
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const PrimaryLinkButton = ({ children, ...otherProps }) => (
	<DynamicLinkButtonComponent linkStyle={primaryLink} {...primary} {...otherProps}>
		{children}
	</DynamicLinkButtonComponent>
);

/**
 * @name OutlinedButton
 * @description Button component to use as primary action when a CTA is already defined.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const OutlinedButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent linkStyle={primaryLink} {...outlined} {...otherProps}>
		{children}
	</DynamicButtonComponent>
);

/**
 * @name OutlinedLinkButton
 * @description Link button component to use as primary when a CTA is already defined.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children The children to display inside this button.
 */
const OutlinedLinkButton = ({ children, ...otherProps }) => (
	<DynamicLinkButtonComponent linkStyle={primaryLink} {...outlined} {...otherProps}>
		{children}
	</DynamicLinkButtonComponent>
);

/**
 * @name StandardButton
 * @description Standard button component to be used for most use-cases.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const StandardButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent {...standard} {...otherProps}>
		{children}
	</DynamicButtonComponent>
);

/**
 * @name SubtleButton
 * @description Button component to be used along a primary button for less crucial actions such as "cancel".
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const SubtleButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent {...subtle} {...otherProps}>
		{children}
	</DynamicButtonComponent>
);

/**
 * @name SubtleLinkButton
 * @description Link button component to be used along a primary button for less used redirections.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const SubtleLinkButton = ({ children, ...otherProps }) => (
	<DynamicLinkButtonComponent linkStyle={subtleLink} {...subtle} {...otherProps}>
		{children}
	</DynamicLinkButtonComponent>
);

/**
 * @name DropdownButton
 * @description Button component to be used for dropdown menu trigger.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const DropdownButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent {...dropdown} {...otherProps}>
		{children}
	</DynamicButtonComponent>
);

/**
 * @name DangerButton
 * @description Button component to use for final confirmation to a destructive action such as deleting data.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const DangerButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent {...danger} {...otherProps}>
		{children}
	</DynamicButtonComponent>
);

/**
 * @name IconButton
 * @description Button component intended to display only the icon it contains, but behave as a regular button.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const IconButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent {...iconButton} {...otherProps} isIconOnly>
		{children}
	</DynamicButtonComponent>
);

/**
 * @name FloatingActionButton
 * @description Button component intended to display a floating action button.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const FloatingActionButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent {...floatingActionButton} {...otherProps}>
		{children}
	</DynamicButtonComponent>
);

PrimaryButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

PrimaryLinkButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

OutlinedButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

OutlinedLinkButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

StandardButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

SubtleButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

SubtleLinkButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

DropdownButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

DangerButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

IconButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

FloatingActionButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

export {
	FloatingActionButton,
	OutlinedButton,
	OutlinedLinkButton,
	PrimaryButton,
	PrimaryLinkButton,
	StandardButton,
	SubtleButton,
	SubtleLinkButton,
	DropdownButton,
	DangerButton,
	IconButton,
};
