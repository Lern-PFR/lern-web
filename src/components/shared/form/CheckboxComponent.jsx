import styled from 'styled-components';
import {
	border,
	display,
	layout,
	shadow,
	color,
} from 'styled-system';
import PropTypes from 'prop-types';
import { disabled as disabledStyle } from 'theme/buttonStyles';
import { checkboxStyle } from 'theme/formStyles';
import { jasmine, tuna } from 'theme/colors';

const StyledCheckboxComponent = styled('input')(
	border,
	display,
	layout,
	shadow,
	color,
	{
		...checkboxStyle,
		appearance: 'none',
		MozAppearance: 'none',
		WebkitAppearance: 'none',
		position: 'relative',
		borderColor: (({ disabled }) => (disabled ? tuna.default : 'initial')),
		backgroundColor: (({ disabled }) => (disabled ? jasmine.darker1 : 'initial')),

		'&:focus': {
			outline: 'none',
		},

		'&::before': {
			position: 'absolute',
			top: 0,
			left: 0,
			content: '',
			fontSize: '12px',
			fontWeight: 700,
			color: 'transparent',
			display: 'inline-block',
			width: '100%',
			height: '100%',
			textAlign: 'center',
			borderColor: (({ disabled }) => (disabled ? tuna.default : 'initial')),
		},

		'&:indeterminate:before': {
			content: '"\\2013"',
			color: 'white',
			backgroundColor: (({ disabled }) => (disabled ? jasmine.darker1 : 'black')),
			borderColor: (({ disabled }) => (disabled ? tuna.default : 'initial')),
		},

		'&:checked:before': {
			content: '"\\2713"',
			color: 'white',
			backgroundColor: (({ disabled }) => (disabled ? jasmine.darker1 : 'black')),
			borderColor: (({ disabled }) => (disabled ? tuna.default : 'initial')),
		},
	},
);

/**
 * @name CheckboxComponent
 * @description A component used to display styled checkbox element.
 *
 * @author Christopher Walker
 *
 * @param {bool}	[disabled]		: Whether the checkbox is disabled.
 * @param {bool}	[checked]		: Whether the checkbox is checked.
 * @param {string}	id				: The id of the checkbox.
 */
const CheckboxComponent = ({ id, checked, disabled, indeterminate, ...otherProps }) => (
	<StyledCheckboxComponent
		{...otherProps}
		{...(disabled ? disabledStyle : {})}
		type="checkbox"
		indeterminate={indeterminate}
		disabled={disabled}
		defaultChecked={checked}
		id={id}
	/>
);

CheckboxComponent.propTypes = {
	id: PropTypes.string.isRequired,
	disabled: PropTypes.bool,
	checked: PropTypes.bool,
	indeterminate: PropTypes.bool,
};

CheckboxComponent.defaultProps = {
	disabled: false,
	checked: false,
	indeterminate: false,
};

export default CheckboxComponent;
