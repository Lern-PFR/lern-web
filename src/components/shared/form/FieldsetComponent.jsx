import styled from 'styled-components';
import {
	border,
	color,
	display,
	flexbox,
	layout,
	margin,
	padding,
	position,
} from 'styled-system';
import { groupStyle } from 'theme/formStyles';
import PropTypes from 'prop-types';
import LegendComponent from './LegendComponent';

const StyledFieldsetComponent = styled('fieldset')(
	border,
	color,
	display,
	flexbox,
	layout,
	margin,
	padding,
	position,
);

/**
 * @name FieldsetComponent
 * @description A component used to display styled fieldset element.
 *
 * @author Christopher Walker
 *
 * @param {string}	title		    : The title of the fieldset.
 * @param {*}	    children		: The child elements of the fieldset.
 */
const FieldsetComponent = ({ children, title, ...otherProps }) => (
	<StyledFieldsetComponent {...groupStyle} {...otherProps}>
		<LegendComponent>
			{title}
		</LegendComponent>
		{children}
	</StyledFieldsetComponent>
);

FieldsetComponent.propTypes = {
	title: PropTypes.string.isRequired,
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

export default FieldsetComponent;
