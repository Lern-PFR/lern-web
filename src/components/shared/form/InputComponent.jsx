import styled from 'styled-components';
import {
	border,
	display,
	layout,
	shadow,
	color,
} from 'styled-system';
import PropTypes from 'prop-types';
import { disabled as disabledStyle } from 'theme/buttonStyles';
import { inputStyle, errorInputStyle } from 'theme/formStyles';
import { forwardRef, useRef } from 'react';

const StyledInputComponent = styled('input')(
	{ ...inputStyle },
	border,
	display,
	layout,
	shadow,
	color,
);

/**
 * @name InputComponent
 * @description A component used to display styled input element.
 *
 * @author Christopher Walker
 *
 * @param {bool}	[disabled]			: Whether the input is disabled.
 * @param {string}	[type]				: The type of input, default is text.
 * @param {string}	[placeholder]		: Placeholder text for this input.
 * @param {bool}	[hasError]			: Whether the input contains a validation error.
 * @param {string}	id					: The id of the input.
 */

const InputComponent = forwardRef(
	({ id, disabled, placeholder, type, hasError, ...otherProps }, ref) => {
		const defaultRef = useRef();
		const resolvedRef = ref || defaultRef;

		return (
			<StyledInputComponent
				{...(disabled ? disabledStyle : {})}
				type={type}
				disabled={disabled}
				placeholder={placeholder}
				id={id}
				ref={resolvedRef}
				{...(hasError ? errorInputStyle : {})}
				{...otherProps}
			/>
		);
	}
);

InputComponent.displayName = 'InputComponent';

InputComponent.propTypes = {
	id: PropTypes.string.isRequired,
	placeholder: PropTypes.string,
	disabled: PropTypes.bool,
	hasError: PropTypes.bool,
	type: PropTypes.string,
};

InputComponent.defaultProps = {
	disabled: false,
	hasError: false,
	type: 'text',
	placeholder: '',
};

export default InputComponent;
