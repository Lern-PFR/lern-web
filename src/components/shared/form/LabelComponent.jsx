import PropTypes from 'prop-types';
import { labelStyle } from 'theme/formStyles';
import DynamicTextComponent from 'components/shared/typography/DynamicTextComponent';
import { canon, greatPrimer, doublePica, bodyCopy, brevier, longPrimer, minion, paragon, pica, trafalgar } from 'theme/textStyles';

/**
 * @name LabelComponent
 * @description A component used to display styled label element.
 *
 * @author Christopher Walker
 *
 * @param {string}	[element]		: The id of the element that the label is associated to.
 * @param {string}	[typography]	: The text style of the label text.
 * @param {string}	children		: The test to be displayed in the label.
 */
const LabelComponent = ({ children, typography, forId, ...otherProps }) => (
	<DynamicTextComponent
		{...(() => {
			switch (typography) {
				case 'canon':		return canon;
				case 'trafalgar':	return trafalgar;
				case 'paragon':		return paragon;
				case 'greatprimer':	return greatPrimer;
				case 'doublepica':	return doublePica;
				case 'bodycopy':	return bodyCopy;
				case 'pica':		return pica;
				case 'longprimer':	return longPrimer;
				case 'minion':		return minion;
				default:			return brevier;
			}
		})()}
		tag="label"
		{...labelStyle}
		htmlFor={forId}
		{...otherProps}
	>
		{children}
	</DynamicTextComponent>
);

LabelComponent.propTypes = {
	children: PropTypes.string.isRequired,
	typography: PropTypes.string,
	forId: PropTypes.string,
};

LabelComponent.defaultProps = {
	typography: 'brevier',
	forId: '',
};

export default LabelComponent;
