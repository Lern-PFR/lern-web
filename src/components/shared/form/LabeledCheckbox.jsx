import { StyledDiv } from 'components/shared/layout';
import PropTypes from 'prop-types';
import { tuna } from 'theme/colors';
import CheckboxComponent from './CheckboxComponent';
import LabelComponent from './LabelComponent';

/**
 * @name LabeledCheckbox
 * @description A component used to display styled checkbox element.
 *
 * @author Christopher Walker
 *
 * @param {bool}	[disabled]		: Whether the checkbox is disabled.
 * @param {bool}	[checked]		: Whether the checkbox is checked.
 * @param {string}	id				: The id of the checkbox.
 * @param {string}	children		: text to display in the label.
 */
const LabeledCheckbox = ({ children, id, checked, disabled, indeterminate, ...otherProps }) => (
	<StyledDiv display="flex" alignItems="center">
		<CheckboxComponent {...otherProps} indeterminate={indeterminate} disabled={disabled} checked={checked} id={id} />
		<LabelComponent forId={id} color={disabled ? tuna.darker1 : 'initial'}>
			{children}
		</LabelComponent>
	</StyledDiv>
);

LabeledCheckbox.propTypes = {
	id: PropTypes.string.isRequired,
	children: PropTypes.string.isRequired,
	disabled: PropTypes.bool,
	checked: PropTypes.bool,
	indeterminate: PropTypes.bool,
};

LabeledCheckbox.defaultProps = {
	disabled: false,
	checked: false,
	indeterminate: false,
};

export default LabeledCheckbox;
