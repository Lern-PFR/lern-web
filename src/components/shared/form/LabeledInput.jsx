import { StyledDiv } from 'components/shared/styledElements';
import PropTypes from 'prop-types';
import { tuna } from 'theme/colors';
import { forwardRef, useRef } from 'react';
import InputComponent from './InputComponent';
import LabelComponent from './LabelComponent';
import SubTextComponent from './SubTextComponent';

/**
 * @name LabeledInput
 * @description A component used to display a styled input element with built-in label, hint and validation options.
 *
 * @author Christopher Walker
 *
 * @param {string}	id					: The id of the input.
 * @param {string}	children			: The text to be displayed in the label.
 * @param {string}	[textStyle]			The text style for this input and its label.
 * @param {string}	[hintText]			: The text to be displayed in the hint.
 * @param {string}	[errorText]			: The text to be displayed in the validation error.
 * @param {bool}	[disabled]			: Whether the input is disabled.
 * @param {string}	[type]				: The type of input, default is text.
 * @param {string}	[placeholder]		: Placeholder text for this input.
 * @param {bool}	[hasError]	: Whether the subtext is an error display or not.
 */
const LabeledInput = forwardRef(
	({ textStyle, id, children, hintText, errorText, disabled, type, placeholder, hasError, ...otherProps }, ref) => {
		const defaultRef = useRef();
		const resolvedRef = ref || defaultRef;
		const errText = hasError ? errorText : '';

		return (
			<StyledDiv display="flex" flexDirection="column">
				<InputComponent textStyle={textStyle} id={id} type={type} disabled={disabled} placeholder={placeholder} ref={resolvedRef} hasError={hasError} {...otherProps} />
				<LabelComponent textStyle={textStyle} order="-1" forId={id} disabled={disabled} hasError={hasError}>
					{children}
				</LabelComponent>
				{(!hasError || (hasError && disabled)) && (
					<SubTextComponent color={disabled ? tuna.darker1 : 'initial'}>
						{hintText}
					</SubTextComponent>
				)}
				{hasError && !disabled && (
					<SubTextComponent isErrorMessage={hasError}>
						{errText}
					</SubTextComponent>
				)}
			</StyledDiv>
		);
	}
);

LabeledInput.displayName = 'LabeledInput';

LabeledInput.propTypes = {
	id: PropTypes.string.isRequired,
	children: PropTypes.string.isRequired,
	textStyle: PropTypes.string,
	hintText: PropTypes.string,
	errorText: PropTypes.string,
	disabled: PropTypes.bool,
	type: PropTypes.string,
	placeholder: PropTypes.string,
	hasError: PropTypes.bool,
};

LabeledInput.defaultProps = {
	textStyle: 'brevier',
	hintText: '',
	errorText: '',
	disabled: false,
	type: undefined,
	placeholder: undefined,
	hasError: false,
};

export default LabeledInput;
