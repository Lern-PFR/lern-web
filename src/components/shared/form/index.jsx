import FieldsetComponent from './FieldsetComponent';
import LabeledCheckbox from './LabeledCheckbox';
import LabeledRadioButton from './LabeledRadioButton';
import LabelComponent from './LabelComponent';
import LabeledInput from './LabeledInput';

export {
	LabeledCheckbox,
	FieldsetComponent,
	LabeledRadioButton,
	LabelComponent,
	LabeledInput,
};
