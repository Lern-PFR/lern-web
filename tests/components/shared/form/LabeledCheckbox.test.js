import { shallow } from 'enzyme';
import { act, cleanup, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { LabeledCheckbox } from 'components/shared/form';

describe('LabeledCheckbox', () => {
	afterEach(() => {
		cleanup();
	});

	it('should match previous snapshot', () => {
		const sut = (<LabeledCheckbox id="lcb">Labeled Checkbox</LabeledCheckbox>);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});

	it('should match previous snapshot for checked state', () => {
		const sut = (<LabeledCheckbox checked id="lcb">Labeled Checkbox</LabeledCheckbox>);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});

	it('should match previous snapshot for disabled state', () => {
		const sut = (<LabeledCheckbox disabled id="lcb">Labeled Checkbox</LabeledCheckbox>);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});

	it('should match previous snapshot for checked and disabled states', () => {
		const sut = (<LabeledCheckbox disabled checked id="lcb">Labeled Checkbox</LabeledCheckbox>);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});

	it('should match previous snapshot for checkbox with custom label text style', () => {
		const sut = (<LabeledCheckbox id="lcb" textStyle="bodycopy">Labeled Checkbox</LabeledCheckbox>);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});

	it('should call the onChange method when clicking on checkbox', async () => {
		const mockedOnChange = jest.fn(() => {});
		const sut = (<LabeledCheckbox onChange={mockedOnChange} id="cb">Labeled Checkbox</LabeledCheckbox>);
		render(sut);

		await act(async () => {
			userEvent.click(screen.getByRole('checkbox'));
		});

		expect(mockedOnChange).toHaveBeenCalled();
	});

	it('should call the onChange method when clicking on label', async () => {
		const mockedOnChange = jest.fn(() => {});
		const sut = (<LabeledCheckbox onChange={mockedOnChange} id="cb">Labeled Checkbox</LabeledCheckbox>);
		render(sut);

		await act(async () => {
			userEvent.click(screen.getByText('Labeled Checkbox'));
		});

		expect(mockedOnChange).toHaveBeenCalled();
	});

	it('should not call the onChange method when clicking on disabled checkbox', async () => {
		const mockedOnChange = jest.fn(() => {});
		const sut = (<LabeledCheckbox onChange={mockedOnChange} disabled id="cb">Labeled Checkbox</LabeledCheckbox>);
		render(sut);

		await act(async () => {
			userEvent.click(screen.getByRole('checkbox'));
		});

		expect(mockedOnChange).not.toHaveBeenCalled();
	});

	it('should not call the onChange method when clicking on disabled label', async () => {
		const mockedOnChange = jest.fn(() => {});
		const sut = (<LabeledCheckbox onChange={mockedOnChange} disabled id="cb">Labeled Checkbox</LabeledCheckbox>);
		render(sut);

		await act(async () => {
			userEvent.click(screen.getByText('Labeled Checkbox'));
		});

		expect(mockedOnChange).not.toHaveBeenCalled();
	});
});
