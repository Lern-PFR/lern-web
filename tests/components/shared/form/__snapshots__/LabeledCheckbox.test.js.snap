// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabeledCheckbox should match previous snapshot 1`] = `
<styled.div
  alignItems="center"
  display="flex"
>
  <CheckboxComponent
    checked={false}
    disabled={false}
    id="lcb"
  />
  <LabelComponent
    color="initial"
    forId="lcb"
    textStyle=""
  >
    Labeled Checkbox
  </LabelComponent>
</styled.div>
`;

exports[`LabeledCheckbox should match previous snapshot for checked and disabled states 1`] = `
<styled.div
  alignItems="center"
  display="flex"
>
  <CheckboxComponent
    checked={true}
    disabled={true}
    id="lcb"
  />
  <LabelComponent
    color="#707070"
    forId="lcb"
    textStyle=""
  >
    Labeled Checkbox
  </LabelComponent>
</styled.div>
`;

exports[`LabeledCheckbox should match previous snapshot for checked state 1`] = `
<styled.div
  alignItems="center"
  display="flex"
>
  <CheckboxComponent
    checked={true}
    disabled={false}
    id="lcb"
  />
  <LabelComponent
    color="initial"
    forId="lcb"
    textStyle=""
  >
    Labeled Checkbox
  </LabelComponent>
</styled.div>
`;

exports[`LabeledCheckbox should match previous snapshot for disabled state 1`] = `
<styled.div
  alignItems="center"
  display="flex"
>
  <CheckboxComponent
    checked={false}
    disabled={true}
    id="lcb"
  />
  <LabelComponent
    color="#707070"
    forId="lcb"
    textStyle=""
  >
    Labeled Checkbox
  </LabelComponent>
</styled.div>
`;
